iocage/lib/Provisioning/ix.py

```diff
@@ -148,19 +178,52 @@ def provision(
     self.jail.fstab.save()
 
     if "pkgs" in pluginDefinition.keys():
-        pkg_packages = " ".join(pluginDefinition["pkgs"])
+        pkg_packages = list(pluginDefinition["pkgs"])
     else:
-        pkg_packages = pluginDefinition.name
+        pkg_packages = [pluginDefinition.name]
 
-    commands = [
-        "ASSUME_ALWAYS_YES=true pkg update",
-        f"ASSUME_ALWAYS_YES=true pkg install -y {pkg_packages}",
-        "./ix-plugin/post_install.sh"
-    ]
-
-    for event in self.jail.fork_exec("\n".join(commands)):
+    for event in self.jail.start():
         yield event
 
+    try:
```

Comment:
Why do a `try: finally` block when any of the `try`'s above the `finally` statement would end execution. I think you can drop a whole nested level here.

Reply:
It would, but instead each nested try block would require the same fallback. This top-level try/finally makes sure that the code in finally is executed independently of the success of its internal steps.